# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/worklet/WorkletImpl.h
# Commit: d801fdeac90b
# Full Hash: d801fdeac90bc63de7f7efd090ced48bf871121a
# Author: Tom Ritter <tom@mozilla.com>
# Date: 2022-11-29 08:40:32
# Regressor Bug: 1778510
# File Overlap Count: 1
# Description:
#   Bug 1778510: Add RTPCallerType to GlobalObject r=asuth
#   
#   This centralizes the logic in one place.
#   
#   In order to do this, we will need to check the principal
# ==============================================================================

diff -r 5af032c628c0 -r d801fdeac90b dom/worklet/WorkletImpl.h
--- a/dom/worklet/WorkletImpl.h	Mon Nov 28 18:04:06 2022 +0000
+++ b/dom/worklet/WorkletImpl.h	Mon Nov 28 18:04:07 2022 +0000
@@ -82,6 +82,7 @@
   const Maybe<nsID>& GetAgentClusterId() const { return mAgentClusterId; }
 
   bool IsSharedMemoryAllowed() const { return mSharedMemoryAllowed; }
+  bool IsSystemPrincipal() const { return mIsSystemPrincipal; }
   bool ShouldResistFingerprinting() const {
     return mShouldResistFingerprinting;
   }
@@ -106,6 +107,8 @@
   ipc::PrincipalInfo mPrincipalInfo;
   // Accessed on only worklet parent thread.
   nsCOMPtr<nsIPrincipal> mPrincipal;
+  // For off-main-thread checks
+  bool mIsSystemPrincipal;
 
   const WorkletLoadInfo mWorkletLoadInfo;
 
